scripts/prepare-for-release

```diff
@@ -37,23 +37,99 @@ FILES_CHANGED=()
 PR_BASE=master  # target
 PR_TITLE="🥑🤖 $LATEST_TAG release prep"
 PR_BODY="🥑🤖 Auto-generated PR for $LATEST_TAG release. Updating release versions in repo."
+PR_LABEL_1="release-prep"
+PR_LABEL_2="🤖 auto-generated🤖"
+
+HELP=$(cat << 'EOM'
+  Update repo with the new release version and create a pr from a new release prep branch.
+  This script prompts the user with complete details about the PR before pushing the new local branch to remote and creating the PR.
+  The new release version is the latest local git tag.
+  Note: The local tag creation for a new release is separated from this script. A new tag must be created before this script is run which is automated when this script is run via make targets. 
+
+  Usage: prepare-for-release [options]
+
+  Options:
+    -d      create a draft pr
+    -r      target repo full name for the pr (default: aws/amazon-ec2-metadata-mock)
+    -h      help
+
+  Examples:
+    prepare-for-release -d                                      update release version in repo and create a draft pr against aws/amazon-ec2-metadata-mock
+    prepare-for-release -r username/amazon-ec2-metadata-mock    update release version in repo and create a pr against username/amazon-ec2-metadata-mock
+EOM
+)
+
+DRAFT=false
+REPO_FULL_NAME=""
+NEED_ROLLBACK=true
+
+process_args() {
+    while getopts "hdr:" opt; do
+        case ${opt} in
+            h )
+            echo -e "$HELP" 1>&2
+            exit 0
+            ;;
+            d )
+            DRAFT=true
+            ;;
+            r )
+            # todo: validate $REPO_FULL_NAME
+            REPO_FULL_NAME="${OPTARG}"
+            ;;
+            \? )
+            echo "$HELP" 1>&2
+            exit 0
+            ;;
+        esac
+    done
+
+    # set repo full name to the default value if unset
+    if [ -z $REPO_FULL_NAME ]; then
+        REPO_FULL_NAME=$DEFAULT_REPO_FULL_NAME
+    fi
+}
+
+# output formatting
+export TERM="xterm"
+RED=$(tput setaf 1)
+MAGENTA=$(tput setaf 5)
+RESET_FMT=$(tput sgr 0)
+BOLD=$(tput bold)
+
+# verify origin tracking before creating and pushing new branches
+verify_origin_tracking() {
+    origin=$(git remote get-url origin 2>&1) || true
+
+    if [[ $origin == "fatal: No such remote 'origin'" ]] || [[ $origin == "https://github.com/aws/amazon-ec2-metadata-mock.git" ]]; then
+        echo -e "❌ ${RED}Expected remote 'origin' to be tracking fork but found \"$origin\". Set it up before running this script again.${RESET_FMT}"
+        NEED_ROLLBACK=false
+        exit 1
+    fi
+}
 
 create_release_branch() {
-    git checkout -b $NEW_BRANCH_NAME
-    echo -e "✅ Created new release branch $NEW_BRANCH_NAME\n\n"
+    exists=$(git checkout -b $NEW_BRANCH 2>&1) || true
+
+    if [[ $exists == "fatal: A branch named '$NEW_BRANCH' already exists." ]]; then
+        echo -e "❌ ${RED}$exists${RESET_FMT}"
+        NEED_ROLLBACK=false
+        exit 1
+    fi
+    echo -e "✅ ${BOLD}Created new release branch $NEW_BRANCH\n\n${RESET_FMT}"
```

Comment:
consider using branch emojis as future enhancement: ᚿ , ᛆ

Reply:
oh! didnt know there is one. will keep this in mind when I touch this file next. :D